src/System/Wlog/LoggerConfig.hs

```diff
@@ -268,19 +270,26 @@ topLevelParams =
 
 instance FromJSON LoggerConfig where
     parseJSON = withObject "rotation params" $ \o -> do
-        _lcRotation        <-                         o .:? "rotation"
-        -- TODO: Support word "All" which will include all severities at once
-        _lcTermSeverityOut <- (fmap Set.fromList) <$> o .:? "termSeveritiesOut"
-        _lcTermSeverityErr <- (fmap Set.fromList) <$> o .:? "termSeveritiesErr"
-        _lcShowTime        <-                 Any <$> o .:? "showTime"    .!= False
-        _lcShowTid         <-                 Any <$> o .:? "showTid"     .!= False
-        _lcFilePrefix      <-                         o .:? "filePrefix"
+        _lcRotation        <-         o .:? "rotation"
+        _lcTermSeverityOut <- parseSeverities o "termSeveritiesOut"
+        _lcTermSeverityErr <- parseSeverities o "termSeveritiesErr"
+        _lcShowTime        <- Any <$> o .:? "showTime"    .!= False
+        _lcShowTid         <- Any <$> o .:? "showTid"     .!= False
+        _lcFilePrefix      <-         o .:? "filePrefix"
         _lcTree            <- parseJSON $ Object $ filterObject topLevelParams o
 
         printConsoleFlag    <- o .:? "printOutput" .!= False
         let _lcConsoleAction = Last $ bool Nothing (Just defaultHandleAction) printConsoleFlag
         let _lcMapper        = mempty
         return LoggerConfig{..}
+          where
+            parseSeverities :: Object -> Text -> Parser (Maybe Severities)
+            parseSeverities o term = do
+                case HM.lookup term o of
+                    Just value -> case value of
+                                String "All" -> pure $ Just allSeverities
+                                Array sevs   -> Just . Set.fromList . Vector.toList <$> Vector.mapM parseJSON sevs
```

Comment:
You should add `_` case here if you don't want to see _Non-exhaustive pattern matching_ erro. I wonder why `-Wall` didn't warned you about that... Rather than that, looks good. You can add `Debug+` and etc. functions :)